Source/WebCore/page/Page.cpp

```diff
@@ -2787,6 +2787,7 @@ void Page::setHeaderHeight(int headerHeight)
     if (!renderView)
         return;
 
+    frameView->availableContentSizeChanged(ScrollableArea::AvailableSizeChangeReason::AreaSizeChanged);
```

Comment:
This isn't strictly correct; "available content size changed" means "the size of the view within which layout occurs changed" i.e. the FrameView resized, and that's not what's happening here; FrameView size does not change, but the size of the scrolled content does.

The correct fix would be to hook up a "header or footer height changed" code path and this would do work similar to what `FrameView::setContentsSize()` does. It might just enough to call `updateScrollbars(scrollPosition())`. We don't need to trigger layout, because the web contents don't change.

Reply:
Fixed. Calling updateScrollbars(scrollPosition()) was enough.